lib/pf/soh.pm

```diff
@@ -258,7 +263,8 @@ sub parse_request {
         }
 
         unless (%ok) {
-            die "Couldn't parse any SoH-MS-Windows-Health-Status lines";
+            $self->{logger}->error("Couldn't parse any SoH-MS-Windows-Health-Status lines");
+            return undef;
```

Comment:
This is wrapped in try {} catch {} block returning in the try block does not exit the function.